src/amo/components/InstallWarning/index.js

```diff
@@ -0,0 +1,184 @@
+/* @flow */
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import type { AppState } from 'amo/store';
+import { ADDON_TYPE_EXTENSION, UNINSTALLED, UNKNOWN } from 'core/constants';
+import translate from 'core/i18n/translate';
+import log from 'core/logger';
+import tracking from 'core/tracking';
+import { withExperiment } from 'core/withExperiment';
+import Notice from 'ui/components/Notice';
+import type { AddonType } from 'core/types/addons';
+import type { I18nType } from 'core/types/i18n';
+import type { InstalledAddon } from 'core/reducers/installations';
+import type { WithExperimentInjectedProps } from 'core/withExperiment';
+
+import './styles.scss';
+
+type Props = {|
+  addon: AddonType,
+|};
+
+type CouldShowWarningParams = {|
+  addonIsExtension: boolean,
+  addonIsRecommended: boolean | void,
+  experimentEnabled: $PropertyType<
+    WithExperimentInjectedProps,
+    'experimentEnabled',
+  >,
+  installStatus: $PropertyType<InstalledAddon, 'status'>,
+|};
+
+export const couldShowWarning = ({
+  addonIsExtension,
+  addonIsRecommended,
+  experimentEnabled,
+  installStatus,
+}: CouldShowWarningParams) => {
+  return (
+    addonIsExtension &&
+    !addonIsRecommended &&
+    experimentEnabled &&
+    [UNINSTALLED, UNKNOWN].includes(installStatus)
```

Comment:
I think you also need `UNINSTALLING`, no?

Reply:
Hmm, I'm not sure. Can an add-on be installed when the current status is `UNINSTALLING`? I would think not, and if it cannot I don't see why we'd need the warning there. What's your use case for needing `UNINSTALLING`?

Reply:
I forgot, heh. Nevermind then, your comment makes a lot of sense, there is no need for `UNINSTALLING`.